server/controller/apis/discussionController.js

```diff
@@ -0,0 +1,378 @@
+/**
+ * Agora - Close the loop
+ * © 2021-2022 Brian Gormanly
+ * BSD 3-Clause License
+ * see included LICENSE or https://opensource.org/licenses/BSD-3-Clause 
+ */
+
+const discussionService = require("../../service/discussionService")
+const ApiMessage = require("../../model/util/ApiMessage");
+
+// Get
+
+exports.getDiscussionByGoalId = async ( req, res ) => {
+
+    // get the user id either from the request user from basic auth in API call, or from the session for the UI
+    let authUserId;
+    if( req.user ) {
+        authUserId = req.user.id;
+    }
+    else if( req.session.authUser ) {
+        authUserId = req.session.authUser.id;
+    }
+
+    const id = req.params.id
+
+    // get the discussion by goal id
+    let discussion = await discussionService.getDiscussion( "goal", id, authUserId );
+
+    if(!discussion) {
+        const message = new ApiMessage( 404, "Not Found", "Discussion not found" );
```

Comment:
If trying to replicate the use of ApiMessage seen in resourceController.js, this should call ApiMessage's createApiMessage function.
```suggestion
        const message = new ApiMessage.createApiMessage( 404, "Not Found", "Discussion not found" );
```

This applies for all uses of ApiMessage in the file, as testing without this declaration provides the following error:
UnhandledPromiseRejectionWarning: TypeError: ApiMessage is not a constructor

Reply:
I guess because it is not exported, rather confusing